### DISCUSSION TITLE
Caused by: com.linecorp.armeria.common.ContentTooLargeException: maxContentLength: 10485760, contentLength: 16377153, transferred: 10491274

### DISCUSSION THREAD

**Discussion by fruitfish** [apache/skywalking#11362] [Q&A] [answered]
skywalking version : 9.5.0
![image](https://github.com/apache/skywalking/assets/19407667/b1b50c79-449d-440b-9f31-39d81dd2b50f)
when I query the trace with the condition "duration", reporting a mistake。
view log
![image](https://github.com/apache/skywalking/assets/19407667/3334cbf0-5919-4bd8-9f85-22a7087972c5)
`Caused by: com.linecorp.armeria.common.ContentTooLargeException: maxContentLength: 10485760, contentLength: 16377153, transferred: 10491274`

I set the config `SW_STORAGE_ES_BATCH_OF_BYTES: 31457280`（default value is 10485760).

![image](https://github.com/apache/skywalking/assets/19407667/b9e607f3-e93e-457f-b2fe-933404263660)

**Comment by fruitfish** [comment]
question continue:
It is still reporting the mistake.   `Caused by: com.linecorp.armeria.common.ContentTooLargeException: maxContentLength: 10485760, contentLength: 16377153, transferred: 10491274` 
Searching  the startup log, the config `SW_STORAGE_ES_BATCH_OF_BYTES` is 31457280. 
![image](https://github.com/apache/skywalking/assets/19407667/dfcc41a8-c1ab-4f4c-bbe7-a87ba6fae138)
How can I resolve this problem?  
@chenhaipeng   @wu-sheng 
https://github.com/apache/skywalking/pull/10287
Thank you!

**Reply by chenhaipeng** [reply, marked as answer]
1. This problem seems to be a limitation of Armeria. The reason is that the return data of response is too large. See the default configuration: [defaultMaxResponseLength](https://javadoc.io/static/com.linecorp.armeria/armeria-javadoc/1.7.2/com/linecorp/armeria/common/Flags.html#defaultMaxResponseLength())


2. `SW_STORAGE_ES_BATCH_OF_BYTES` is to solve the problem of a single flush batch being too large during ES insertion. For example, if you insert >10M each time, ES may reject your request;

3. How to solve your problem? Query in kibana of ES to confirm whether your data return is too large. If so, it is recommended to streamline the reporting or temporarily adjust the limitations of the Armeria client.

**Reply by wu-sheng** [reply]
@chenhaipeng Although I think 10m is a reasonable length, do you ever face responses over 10m?

**Reply by chenhaipeng** [reply]
Not yet, 10M is also the default value of armeria defaultMaxRequestLength and defaultMaxResponseLength

**Reply by wu-sheng** [reply]
OK. I am not sure whether we should consider to expose this configuration?

**Reply by fruitfish** [reply]
I  temporarily adjust the limitations of the Armeria client. It works!  thank you very much!
And how can I streamline the reporting? Is the trace message too large? so When I query  , the message returned is too large.
 ,
